cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1174,13 +1208,52 @@ public MapAndStatus(T loadMap, K status) {
      */
     private KafkaRebalanceAnnotation rebalanceAnnotation(Reconciliation reconciliation, KafkaRebalance kafkaRebalance) {
         String rebalanceAnnotationValue = rawRebalanceAnnotation(kafkaRebalance);
+        KafkaRebalanceState state = state(kafkaRebalance);
         KafkaRebalanceAnnotation rebalanceAnnotation;
         try {
             rebalanceAnnotation = rebalanceAnnotationValue == null ?
                     KafkaRebalanceAnnotation.none : KafkaRebalanceAnnotation.valueOf(rebalanceAnnotationValue);
+            if (state != null) {
```

Comment:
maybe this code could be simplified a lot. We can do the other way around ... switch on the annotation because we are interested in just three values `approve`, `stop` and `refresh`, the default would be just returning an  `unknown`.
On each annotation case, you can just filter the right (or wrong) state with a single if and returning `unknown` in case, otherwise the current annotation.

Reply:
I think Paolo's suggestion is a good one. It would be better to switch on the annotation.